analyzers/src/SonarAnalyzer.Common/Helpers/SymbolHelper.cs

```diff
@@ -263,19 +264,68 @@ private static bool IsAnyAttributeInOverridingChain<TSymbol>(TSymbol symbol, Fun
             symbol switch
             {
+                { Kind: SymbolKind.Alias } => "alias",
+                { Kind: SymbolKind.ArrayType } => "array",
+                { Kind: SymbolKind.Assembly } => "assembly",
+                { Kind: SymbolKindEx.Discard } => "discard",
+                { Kind: SymbolKind.DynamicType } => "dynamic",
+                { Kind: SymbolKind.ErrorType } => "error",
                 { Kind: SymbolKind.Event } => "event",
+                { Kind: SymbolKindEx.FunctionPointerType } => "function pointer",
                 { Kind: SymbolKind.Field } => "field",
-                { Kind: SymbolKind.Method } => "method",
+                { Kind: SymbolKind.Label } => "label",
+                { Kind: SymbolKind.Local } => "local",
+                { Kind: SymbolKind.Namespace } => "namespace",
+                { Kind: SymbolKind.NetModule } => "netmodule",
+                { Kind: SymbolKind.PointerType } => "pointer",
+                { Kind: SymbolKind.Preprocessing } => "preprocessing",
+                { Kind: SymbolKind.Parameter } => "parameter",
+                { Kind: SymbolKind.RangeVariable } => "range variable",
                 { Kind: SymbolKind.Property } => "property",
-                INamedTypeSymbol { TypeKind: TypeKind.Delegate } => "delegate",
-                INamedTypeSymbol { TypeKind: TypeKind.Class } namedType => namedType.IsRecord() ? "record" : "class",
-                INamedTypeSymbol { TypeKind: TypeKind.Struct } namedType => namedType.IsRecord() ? "record struct" : "struct",
-                INamedTypeSymbol { TypeKind: TypeKind.Enum } => "enum",
+                { Kind: SymbolKind.TypeParameter } => "type parameter",
+                IMethodSymbol methodSymbol => methodSymbol switch
+                {
+                    { MethodKind: MethodKind.BuiltinOperator or MethodKind.UserDefinedOperator or MethodKind.Conversion} => "operator",
+                    { MethodKind: MethodKind.Constructor or MethodKind.StaticConstructor or MethodKind.SharedConstructor } => "constructor",
+                    { MethodKind: MethodKind.Destructor } => "destructor",
+                    { MethodKind: MethodKind.PropertyGet } => "getter",
+                    { MethodKind: MethodKind.PropertySet } => "setter",
+                    _ => "method",
+                },
+                INamedTypeSymbol namedTypeSymbol => namedTypeSymbol switch
+                {
+                    { TypeKind: TypeKind.Array } => "array",
+                    { TypeKind: TypeKind.Class } namedType => namedType.IsRecord() ? "record" : "class",
+                    { TypeKind: TypeKind.Dynamic } => "dynamic",
+                    { TypeKind: TypeKind.Delegate } => "delegate",
+                    { TypeKind: TypeKind.Enum } => "enum",
+                    { TypeKind: TypeKind.Error } => "error",
+                    { TypeKind: (TypeKind)13 } => "function pointer",
```

Comment:
I suggest creating a static class (`TypeKindEx`) with constant values for this TypeKind. Just like there is `SymbolKindEx`, `SyntaxKindEx` and `MethodKindEx`. This way it can be reused in the future.